lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -65,14 +65,34 @@ sub register {
   $self;
 }
 
-sub _add_default_response {
-  my ($self, $op_spec, $code) = @_;
-  return if $op_spec->{responses}{$code};
-  my $name   = $self->{default_response_name};
-  my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
-  my %schema = ('$ref' => "#/definitions/$name");
+sub _add_default_response_2 {
+  my ($self, $op_spec) = @_;
+  for my $code (@{$self->{default_response_codes}}) {
+    next if $op_spec->{responses}{$code};
+
+    my $name   = $self->{default_response_name};
+    my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
+    my %schema = ('$ref' => "#/definitions/$name");
+    tie %schema, 'JSON::Validator::Ref', $ref, $schema{'$ref'}, $schema{'$ref'};
```

Comment:
Do you need to do the four lines over and over again inside the loop? This is not a question about optimisation/speed, but more because it makes me question "why" when I read the code. I'm guessing it's because I already had it that way in my code maybe? If so, then just move them outside the loop.

Reply:
You were right. I just reproduced original code here. Fixed it.